go/client/cmd_nstatus.go

```diff
@@ -0,0 +1,214 @@
+	UserID                 string
+	DeviceID               string
+	DeviceName             string
+	DeviceStatus           string
+	LoggedInProvisioned    bool `json:"LoggedIn"`
+	PassphraseStreamCached bool `json:"KeychainUnlocked"`
+	SessionStatus          string
+	ConfigPath             string
+
+	Client struct {
+		Version string
+	}
+	Service struct {
+		Version string
+		Running bool
+		Pid     string
+		Log     string
+	}
+	KBFS struct {
+		Version string
+		Running bool
+		Pid     string
+		Log     string
+	}
+	Desktop struct {
+		Running bool
+	}
+
+	DefaultUsername      string
+	ProvisionedUsernames []string
+}
+
+func (c *CmdNStatus) Run() error {
+	status, err := c.load()
+	if err != nil {
+		return err
+	}
+
+	return c.output(status)
+}
+
+func (c *CmdNStatus) load() (*fstatus, error) {
+	var status fstatus
+
+	status.Client.Version = libkb.VersionString()
+
+	cli, err := GetConfigClient(c.G())
+	if err != nil {
+		return nil, err
+	}
+
+	curStatus, err := cli.GetCurrentStatus(context.TODO(), 0)
+	if err != nil {
+		return nil, err
+	}
+
+	status.LoggedInProvisioned = curStatus.LoggedIn
+	if curStatus.User != nil {
+		status.Username = curStatus.User.Username
+		status.UserID = curStatus.User.Uid.String()
+	}
+
+	extStatus, err := cli.GetExtendedStatus(context.TODO(), 0)
+	if err != nil {
+		return nil, err
+	}
+
+	config, err := cli.GetConfig(context.TODO(), 0)
+	if err != nil {
+		return nil, err
+	}
+
+	status.ConfigPath = config.ConfigPath
+	status.Service.Version = config.Version
+
+	status.DeviceID = extStatus.DeviceID.String()
+	status.DeviceName = extStatus.DeviceName
+	status.DeviceStatus = extStatus.DeviceStatus
+
+	if extStatus.Standalone {
+		status.Service.Running = false
+	} else {
+		status.Service.Running = true
+		status.Service.Log = path.Join(extStatus.LogDir, c.serviceLogFilename())
+	}
+
+	status.SessionStatus = extStatus.SessionStatus
+	status.PassphraseStreamCached = extStatus.PassphraseStreamCached
+
+	kbfsVersion, err := install.KBFSBundleVersion(c.G(), "")
+	if err == nil {
+		status.KBFS.Version = kbfsVersion
+	}
+	status.KBFS.Log = path.Join(extStatus.LogDir, c.kbfsLogFilename())
+
+	status.Desktop.Running = extStatus.DesktopUIConnected
+
+	status.DefaultUsername = extStatus.DefaultUsername
+	status.ProvisionedUsernames = extStatus.ProvisionedUsernames
+
+	// set anything os-specific:
+	if err := c.osSpecific(&status); err != nil {
+		return nil, err
+	}
+
+	return &status, nil
+}
+
+func (c *CmdNStatus) output(status *fstatus) error {
+	if c.json {
+		return c.outputJSON(status)
+	}
+
+	return c.outputTerminal(status)
+}
+
+func (c *CmdNStatus) outputJSON(status *fstatus) error {
+	b, err := json.MarshalIndent(status, "", "    ")
+	if err != nil {
+		return err
+	}
+	dui := c.G().UI.GetDumbOutputUI()
+	_, err = dui.Printf(string(b) + "\n")
+	return err
+}
+
+func (c *CmdNStatus) outputTerminal(status *fstatus) error {
+	dui := c.G().UI.GetDumbOutputUI()
+	dui.Printf("Username:      %s\n", status.Username)
```

Comment:
you mean the UI-related helper in libkb :smile: ?

I tried with tabs, but the data started to get far from the headings.